ios/Classes/SwiftFlutterVlcPlayerPlugin.swift

```diff
@@ -0,0 +1,284 @@
+                        self.player.pause()
+                    } else if (playbackState == "stop") {
+                        self.player.stop()
+                    }
+                    
+                    result(nil)
+                    return
+                case .dispose:
+                    self.player.stop()
+                    return
+                case .changeURL:
+                    self.player.stop()
+                    let url = arguments["url"] as? String
+                    let media = VLCMedia(url: URL(string: url!)!)
+                    self.player.media = media
+                    result(nil)
+                    return
+                    
+                case .getSnapshot:
+                    let drawable:UIView = self.player.drawable as! UIView
+                    let size = drawable.frame.size
+                    
+                    UIGraphicsBeginImageContextWithOptions(size , _: false, _: 0.0)
+                    
+                    let rec = drawable.frame
+                    drawable.drawHierarchy(in: rec , afterScreenUpdates: false)
+                    
+                    let image = UIGraphicsGetImageFromCurrentImageContext()
+                    UIGraphicsEndImageContext()
+                    
+                    let byteArray = (image ?? UIImage()).pngData()
+                    
+                    result([
+                        "snapshot": byteArray
+                    ])
+                    return
+                    
+                case .setPlaybackSpeed:
+                    
+                    let playbackSpeed = arguments["speed"] as? String
+                    let rate = (playbackSpeed! as NSString).floatValue
+                    self.player.rate = rate
+                    result(nil)
+                    return
+                    
+                case .setTime:
+                    
+                    let time = VLCTime(number: arguments["time"] as? NSNumber)
+                    self.player.time = time
+                    result(nil)
+                    return
+                    
+                default:
+                    result(FlutterMethodNotImplemented)
+                    return
+                }
+            }
+            
+        })
+        
+        eventChannel.setStreamHandler(eventChannelHandler)
+        return hostedView
+        
+    }
+    
+    
+    
+}
+
+class VLCPlayerEventStreamHandler:NSObject, FlutterStreamHandler, VLCMediaPlayerDelegate {
+    
+    private var eventSink: FlutterEventSink?
+    
+    
+    func onListen(withArguments arguments: Any?, eventSink events: @escaping FlutterEventSink) -> FlutterError? {
+        self.eventSink = events
+        return nil
+    }
+    
+    func onCancel(withArguments arguments: Any?) -> FlutterError? {
+        eventSink = nil
+        return nil
+    }
+    
+    
+    func mediaPlayerStateChanged(_ aNotification: Notification?) {
```

Comment:
when you does not have an `eventSink` to send the message, you basically do nothing. So, what do you think about doing 
```
guard let eventSink = self.eventSink else { return }
```